### PR TITLE
Allow to include multiple appIds to filter EventTypes

### PR DESCRIPTION
**Pull request by josejulio**

### CODE REVIEW

**Comment by burmanm** [comment]
This isn't very descriptive, could you open the reason or what filterOptions means? As in, why is it important in the URL and with ``label`` and ``value`` ?

File: src/main/java/com/redhat/cloud/notifications/db/ApplicationResources.java [hunk 1]
```diff
@@ -146,13 +147,13 @@
         return this.getEventTypes(limiter, null);
     }
 
-    public Multi<EventType> getEventTypes(Query limiter, UUID applicationId) {
+    public Multi<EventType> getEventTypes(Query limiter, List<UUID> applicationId) {
```
Comment (burmanm) [inline src/main/java/com/redhat/cloud/notifications/db/ApplicationResources.java:150]: Perhaps Set<UUID> instead?

File: src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java [hunk 2]
```diff
@@ -91,7 +92,7 @@ private String getAddress(Principal principal) {
     @GET
     @Path("/eventTypes")
     @RolesAllowed("read")
-    public Multi<EventType> getEventTypes(@BeanParam Query query, @QueryParam("applicationId") UUID applicationId) {
+    public Multi<EventType> getEventTypes(@BeanParam Query query, @QueryParam("applicationId") List<UUID> applicationId) {
```
Comment (burmanm) [inline src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:95]: Maybe this should be renamed to "applicationIds" or something to indicate it takes a list?

**Comment by josejulio** [comment]
Done !

File: src/main/java/com/redhat/cloud/notifications/db/ApplicationResources.java [hunk 3]
```diff
@@ -146,13 +147,13 @@
...
-    public Multi<EventType> getEventTypes(Query limiter, UUID applicationId) {
+    public Multi<EventType> getEventTypes(Query limiter, Set<UUID> applicationId) {
         String basicQuery = "SELECT et.id AS et_id, et.name AS et_name, et.description AS et_desc, a.id AS a_id, a.name AS a_name, a.description as a_description FROM public.event_type et " +
                 "JOIN public.application_event_type aet ON aet.event_type_id = et.id " +
                 "JOIN public.applications a ON a.id = aet.application_id";
 
-        if (applicationId != null) {
-            basicQuery += " WHERE a.id = $1";
+        if (applicationId != null && applicationId.size() > 0) {
+            basicQuery += " WHERE a.id = ANY ($1)";
```
Comment (pilhuhn) [inline src/main/java/com/redhat/cloud/notifications/db/ApplicationResources.java:156]: `= ANY` confused me for a moment as I would have used `IN` :)

**Review by pilhuhn — APPROVED** [review]

File: src/main/java/com/redhat/cloud/notifications/db/ApplicationResources.java (hunk 3 above)
Comment (josejulio) [reply src/main/java/com/redhat/cloud/notifications/db/ApplicationResources.java:156]: I tried `IN` initially, but didn't work (the query failed).

**Pull request merged**